pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```diff
@@ -1327,7 +1328,7 @@ type Source struct {
 	httpClient *http.Client
 }
 
-func (s *Source) compactAndUnNullVariables(input []byte) []byte {
+func (s *Source) compactAndUnNullVariables(input []byte, undefinedVariables ...string) []byte {
```

Comment:
can we just make `undefinedVariables ` a slice of string instead of variadic?

Reply:
sure, we will need to update all the data source tests after that

Reply:
ok